cub/cub/device/device_transform.cuh

```diff
@@ -51,31 +52,42 @@ struct DeviceTransform
   //! types must be convertible to the parameters of the function object's call operator. The return type of the call
   //! operator must be assignable to the dereferenced output iterator.
   //! @param stream **[optional]** CUDA stream to launch kernels within. Default is stream\ :sub:`0`.
-  template <typename... RandomAccessIteratorsIn, typename RandomAccessIteratorOut, typename TransformOp>
+  template <typename... RandomAccessIteratorsIn, typename RandomAccessIteratorOut, typename TransformOp, typename NumItemsT>
   CUB_RUNTIME_FUNCTION static cudaError_t Transform(
     ::cuda::std::tuple<RandomAccessIteratorsIn...> inputs,
     RandomAccessIteratorOut output,
-    int num_items,
+    NumItemsT num_items,
     TransformOp transform_op,
     cudaStream_t stream = nullptr)
   {
     CUB_DETAIL_NVTX_RANGE_SCOPE("cub::DeviceTransform::Transform");
+
+    using ChooseOffsetT = detail::choose_signed_offset<NumItemsT>;
+    using OffsetT       = typename ChooseOffsetT::type;
```

Comment:
I thought that `snake_case` only applies to our detail namespace?

Reply:
AFAIK it applies everywhere except for template parameters.

Reply:
The exceptions are template parameters and API conventions in CUB ( `cub::DeviceTransform`).  Since the names here are implementation detail, I'd go with snake case.